src/JMSSerializerAdapter.php

```diff
@@ -0,0 +1,209 @@
+                    'class' => \is_object($data) ? \get_class($data) : \gettype($data),
+                    'exception' => $t,
+                ]);
+            }
+        }
+
+        return $this->originalSerializer->serialize($data, $format, $context);
+    }
+
+    public function deserialize($data, $type, $format, JMSDeserializationContext $context = null)
+    {
+        if ('json' === $format && $this->useGeneratedDeserializer($type, $context)) {
+            try {
+                return $this->liipSerializer->deserialize($data, $type, $format, $this->createLiipContext($context));
+            } catch (\Throwable $t) {
+                $this->logger->warning('Liip Serializer failed to deserialize to type {type}, falling back to JMS', [
+                    'type' => $type,
+                    'exception' => $t,
+                ]);
+            }
+        }
+
+        return $this->originalSerializer->deserialize($data, $type, $format, $context);
+    }
+
+    public function toArray($data, JMSSerializationContext $context = null)
+    {
+        if ($this->useGeneratedSerializer($data, $context)) {
+            try {
+                return $this->liipSerializer->toArray($data, $this->createLiipContext($context));
+            } catch (\Throwable $t) {
+                $this->logger->warning('Liip Serializer failed to convert {type} to array, falling back to JMS', [
+                    'class' => \is_object($data) ? \get_class($data) : \gettype($data),
+                    'exception' => $t,
+                ]);
+            }
+        }
+
+        return $this->originalSerializer->toArray($data, $context);
+    }
+
+    public function fromArray(array $data, $type, JMSDeserializationContext $context = null)
+    {
+        if ($this->useGeneratedDeserializer($type, $context)) {
+            try {
+                return $this->liipSerializer->fromArray($data, $type, $this->createLiipContext($context));
+            } catch (\Throwable $t) {
+                $this->logger->warning('Liip Serializer failed to create {type} from array, falling back to JMS', [
+                    'type' => $type,
+                    'exception' => $t,
+                ]);
+            }
+        }
+
+        return $this->originalSerializer->fromArray($data, $type, $context);
+    }
+
+    private function createLiipContext(?JMSContext $context)
+    {
+        if (null === $context) {
+            return null;
+        }
+
+        $liipContext = new Context();
+        if ($context->hasAttribute('groups')) {
+            $liipContext->setGroups($context->getAttribute('groups'));
+        }
+        if ($context->hasAttribute('version')) {
+            $liipContext->setVersion($context->getAttribute('version'));
+        }
+
+        return $liipContext;
+    }
+
+    private function useGeneratedSerializer($data, ?JMSSerializationContext $context): bool
```

Comment:
For this function and the one below, maybe use "LiipSerializer" instead of "GeneratedSerializer" to be explicit.

Reply:
uh good catch, this was a leftover from the old naming. fixed.